File: csharp/1009.complement-of-base-10-integer.cs
/*
 * @lc app=leetcode id=1009 lang=csharp
 *
 * [1009] Complement of Base 10 Integer
 */

// @lc code=start
public class Solution {
    public int BitwiseComplement(int N) {
        if (N == 0) {
            return 1;
        }
        int l;
        for (l = 31; l >= 0 && (N & (1 << l)) == 0; --l);
        return N ^ ((1 << (l + 1)) - 1);
    }
}
// @lc code=end

File: java/1009.complement-of-base-10-integer.java
/*
 * @lc app=leetcode id=1009 lang=java
 *
 * [1009] Complement of Base 10 Integer
 */

// @lc code=start
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) {
            return 1;
        }
        int ret = 0, k = 0;
        while (n != 0) {
            ret |= (1 & (1 ^ n)) << k;
            n >>= 1;
            ++k;
        }
        return ret;
    }
}
// @lc code=end

